Tool/SuperCAT.ps1

```diff
@@ -1,398 +1,359 @@
-	New-Item -ItemType Directory -Path "$AVLogs" | Out-Null
-$SCAPLogs     = "$Drive\..\..\Outputs\SCAPLogs"
-	New-Item -ItemType Directory -Path "$SCAPLogs" | Out-Null
-$EventLogs    = "$Drive\..\..\Outputs\EventLogs"
-	New-Item -ItemType Directory -Path "$EventLogs" | Out-Null
-
-$Date         = Get-Date -Format "yy-MM-dd"
-$Win32OS      = Get-WMIObject -Class Win32_OperatingSystem
-$ComputerName = $Win32OS.PSComputerName
-$OSArch       = $Win32OS.OSArchitecture
-###########################################################################################################
-
-###########################################################################################################
-## This snippet of code checks to see if the script is being ran as an administrator. If not, it notifies
-## the user that it may not work as intended.
-###########################################################################################################
-
-$currentPrincipal = New-Object Security.Principal.WindowsPrincipal([Security.Principal.WindowsIdentity]::GetCurrent())
-if(($currentPrincipal.IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) -eq $False){
-    Write-Warning "THIS SCRIPT WAS NOT RUN AS AN ADMINISTRATOR! SOME TASKS MAY NOT WORK OR PROVIDE INACCURATE RESULTS!"
-}
-
-###########################################################################################################
-## The user will provide a number or numbers (0-7) and the script writes it to an array. For each number, a
-## specific task sequence will be executed. If 6 is used, all tasks are performed.
-###########################################################################################################
-Write-Output "
-===============================================================
-Select from the following options, inputting only numbers and
-commas (i.e. 1,3,4,5):
-
-0 = Update Antivirus (Requires Pre-Approved Actions)
-
-1 = Collect Computer Information
-2 = Initialize Antivirus Scan
-3 = Collect Antivirus Logs
-4 = Initialize SCAP
-5 = Collect Windows Event Logs
-6 = All Tasks (Collection Only, No Antivirus Updating)
-
-7 = Exit Program
-==============================================================="
-if($PSBoundParameters.ContainsKey("Choices")){
-    $Choices = $PSBoundParameters.Choices
-    Write-Output $Choices
-}
-else{
-    $Choices = Read-Host
-}
+<###############################################################################
+## SUPERCAT (CYBER ASSESSMENT TOOL) V0.3.4
+## DEVELOPED BY: SSGT CLINTON REEL // CLINTON.REEL@US.AF.MIL
+## ADDITIONAL DEVELOPERS IN CONTRIBUTORS.TXT
+###############################################################################>
 
-while($Choices -notmatch "^[0-7,]*$"){
-    $Choices = Read-Host "Please only input numbers 0 to 7 and commas (i.e. 1,3,5)"
-}
+######################### Module Imports #######################################
+using namespace System.Collections.Generic

```

Comment:
**Message from PSScryptAnalyzer**
[{
	"resource": "/FredericIV/SuperCAT/Tool/SuperCAT.ps1",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": "PSUseCompatibleSyntax",
	"severity": 8,
	"message": "The using statement syntax 'using ...;' is not available by default in PowerShell versions 3,4",
	"source": "PSScriptAnalyzer",
	"startLineNumber": 10,
	"startColumn": 1,
	"endLineNumber": 10,
	"endColumn": 43
}]

Reply:
Removed System.Collections.Generic dependency. Had to wrangle some weird typecasting bugs, but it's done.